### PR TITLE
Add API docs for the package details template resource

### PR DESCRIPTION
**Pull request by joelverhagen**
Progress on https://github.com/NuGet/Home/issues/5299.

Client PR: https://github.com/NuGet/NuGet.Client/pull/2518
Service index PR: https://github.com/NuGet/NuGet.Services.Index/pull/132

### CODE REVIEW

**Comment by karann-msft** [comment]
OPS Build status updates of commit _[33c0bed](https://github.com/NuGet/docs.microsoft.com-nuget/pull/1336/commits/33c0bed2b0606345bb6227084625d76ca1a8b92a)_: 

:clock10: **Preparing**: average preparing time is _41 sec(s)_

**Comment by karann-msft** [comment]
OPS Build status updates of commit _[33c0bed](https://github.com/NuGet/docs.microsoft.com-nuget/pull/1336/commits/33c0bed2b0606345bb6227084625d76ca1a8b92a)_: 

:clock10: **Incremental building**: average incremental building time is _21 sec(s)_

**Comment by karann-msft** [comment]
OPS Build status updates of commit _[33c0bed](https://github.com/NuGet/docs.microsoft.com-nuget/pull/1336/commits/33c0bed2b0606345bb6227084625d76ca1a8b92a)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/TOC.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/jver-pdurl/docs/TOC.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/api/package-details-template-resource?branch=pr-en-us-1336) |
[docs/api/overview.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/jver-pdurl/docs/api/overview.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/api/overview?branch=pr-en-us-1336) |
[docs/api/package-details-template-resource.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/jver-pdurl/docs/api/package-details-template-resource.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/api/package-details-template-resource?branch=pr-en-us-1336) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C3%5C2%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201903020059432772-1336%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=5BwGLWi63UFBAPaGlJTdvn1SPsCtpMEtjiz3qXX79x0%3D&st=2019-03-02T00%3A55%3A50Z&se=2019-04-02T01%3A00%3A50Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/api/package-details-template-resource.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+### URL placeholders
+
+Name        | Type    | Required | Notes
+----------- | ------- | -------- | -----
+`{id}`      | string  | no       | The package ID to get details for
+`{version}` | string  | no       | The package version to get details for
+
+The `{id}` and `{version}` values interpreted by the server implementation must be case insensitive and not sensitive to
+whether the version is normalized. In other words, the server should accept the package ID with any casing and
```
Comment (loic-sharma) [inline docs/api/package-details-template-resource.md]: This paragraph gets the point across but feels a little awkward. Also, the second sentence doesn't restate that the version is also case insensitive. How about something like:

> The server should accept `{id}` and `{version}` values with any casing. In addition, the server not be sensitive to whether the version is normalized. In other words, the server should accept also accept non-normalized versions.

Also, could we link to something that explains what normalized versions mean? Something like: https://docs.microsoft.com/en-us/nuget/reference/package-versioning#normalized-version-numbers

Comment (joelverhagen) [reply docs/api/package-details-template-resource.md]: Sure, sounds good.

**Pull request merged**